fraos/src/seqno.rs

```diff
@@ -39,45 +55,101 @@ impl SeqNoIndex {
     }
 
     /// Get the location of a record with the given number.
-    pub fn get_pointer_to_value(&self, seqno: usize) -> Option<u64> {
-        let offset = seqno * size_of::<u64>();
+    pub fn get_offset_and_length(
+        &self,
+        seqno: usize,
+    ) -> Result<Option<(usize, usize)>, FraosError> {
+        let offset = seqno * Self::SIZE_OF_USIZE * 2;
+
+        match self.inner.get_data(offset, |mmap| {
+            let mut offset_buffer = [0u8; Self::SIZE_OF_USIZE];
+            let mut length_buffer = [0u8; Self::SIZE_OF_USIZE];
+            offset_buffer.copy_from_slice(&mmap[..Self::SIZE_OF_USIZE]);
+            length_buffer.copy_from_slice(&mmap[Self::SIZE_OF_USIZE..Self::SIZE_OF_USIZE * 2]);
+
+            Ok(Some((
+                usize::from_le_bytes(offset_buffer),
```

Comment:
why not just use `u64` or something instead of `usize`? This means a 32bits machine wouldn't be able to read the file. Of course those are not really used that much nowadays, but it's still a limitation. An a 128bits pointer machine probably wouldn't need that much space anyway, and it still won't be able to read it.

Reply:
good point, u64 can be a good option